pkg/action/utils.go

```diff
@@ -0,0 +1,42 @@
+package action
+
+import (
+	"helm.sh/helm/v3/pkg/chart"
+)
+
+// SetNamespace sets the Namespace that should be used based on annotations or fallback to default
+// both on the action and on the storage.
+// if setDefault is true then it will just set the default namespace
+// This will read the chart annotations. If no annotations, it leave the existing ns in the action.
+func SetNamespace(x interface{}, chart *chart.Chart, defaultns string, setDefault bool) {
+	namespace := getNamespace(chart, defaultns)
+	// setDefault is mainly used when we set the namespace via the cli flag --namespace and it has priority over
+	// everything else
+	if setDefault {
+		namespace = defaultns
+	}
+
+	switch i := x.(type) {
+	case *Install:
+		i.Namespace = namespace
+		i.Config.SetNamespace(namespace)
```

Comment:
I could probably move this line to the last line after the case to avoid duplicating it, as we will run it always.

Or even remove the whole switch...case as my projection was to just call the `SetNamespace` on every action and then filter here instead, but it doesn't make much sense, does it?

Reply:
well it may protect if we call `SetNamespace` from an action that has no `Namespace` as go wont catch that due to accepting an `interface` so it cant infer if it has a `Namespace` attribute....I think?????

Reply:
I dont think we can call this from an action that has no `Namespace` so I think the whole switch...case needs to go.

Reply:
ah seems that unless we do this, go will complain  of:

```
pkg/action/utils.go:44:3: x.Namespace undefined (type interface {} is interface with no methods)
pkg/action/utils.go:45:3: x.Config undefined (type interface {} is interface with no methods)

```

Reply:
This is some really confusing behaviour of go... I do not really like it, but I have no alternatives I could think of.